src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -310,16 +322,46 @@
     <MakeDir Directories="$(ProjectSpecificConfDir)" />
   </Target>
 
+    <Target Name="InvokeSonarWriteProjectDataNonRazorCompilation"
+          DependsOnTargets="SonarCategoriseProject;SonarWriteFilesToAnalyze;SonarCreateProjectSpecificDirs"
+          AfterTargets="Build"
+          Condition=" $(SonarQubeTempPath) != '' AND '@(RazorCompile)'=='' ">
+      <CallTarget Targets="SonarWriteProjectData" />
+    </Target>
+
+    <Target Name="InvokeSonarWriteProjectDataRazorCompilation"
+          DependsOnTargets="SonarCategoriseProject;SonarWriteFilesToAnalyze;SonarCreateProjectSpecificDirs"
+          BeforeTargets="SonarPrepareRazorCodeAnalysis"
+          AfterTargets="CoreCompile"
+          Condition=" $(SonarQubeTempPath) != '' AND '@(RazorCompile)'!='' ">
+      <CallTarget Targets="SonarWriteProjectData" />
+    </Target>
+
   <!-- **************************************************************************** -->
   <!-- Write the project info file for the current project.
        Any targets which affect @(AnalysisResults) should be run before this target,
        otherwise those results will not appear in the project info file.
   -->
   <!-- **************************************************************************** -->
-  <Target Name="SonarWriteProjectData"
-          DependsOnTargets="SonarCategoriseProject;SonarWriteFilesToAnalyze;SonarCreateProjectSpecificDirs"
-          AfterTargets="Build"
-          Condition=" $(SonarQubeTempPath) != '' ">
+  <Target Name="SonarWriteProjectData" >
```

Comment:
Line 342 and 343 above are obsolete now. That makes line 341 unneeded and we can remove the whole comment section, including **************************